pkg/crc/image/image.go

```diff
@@ -107,6 +107,8 @@ func getImageName(preset crcpreset.Preset) string {
 		return "podman-bundle"
 	case crcpreset.OKD:
 		return "okd-bundle"
+	case crcpreset.Microshift:
+		return "microshift-bundle"
```

Comment:
We have code in pkg/crc/constants.go which goes from preset to bundle URL. This code is used in pkg/crc/machine/bundle/metadata.go to get the default bundle https URLs. The same file also has some code to deal with bundles stored in registry, and in this case it relies on a `""` magic string to indicate to use the default bundle registry URL for the bundle, and it's code in pkg/crc/image/ which does the preset -> default bundle mapping.

Can we use this PR as an opportunity to unify this behaviour a little bit? (remove the magic `""` argument to `image.Pull()`, and move this logic to pkg/crc/constants instead, and make use of this in pkg/crc/machine/bundle)

Reply:
If I understood correctly what you want that logic of getting defaultURI for Quay should be part of `pkg/crc/contants` and should be consume in the `pkg/crc/machine/bundle/metadata.go`  ?

So that `image.PullBundle(preset, "")`  become `image.PullBundle(constants.GetDefaultBundleImageRegistryURL(preset))` ? If we need to do that then we need to move `getImageName` and `getPresetNameE` helper from `/pkg/crc/image/image.go` to `constants.go`.

Reply:
Yes, my suggestion is to try something like this, and to see if it makes the code easier to follow/better organized or not.

Reply:
@cfergeau I can try that in a separate refactor PR, can we just try to focus on quay logic in this PR.

Reply:
https://github.com/crc-org/crc/pull/3817

Reply:
> @cfergeau I can try that in a separate refactor PR, can we just try to focus on quay logic in this PR.

I already brought this up when the quay support was introduced, at the time the answer was already in the same line ;) we cannot keep adding technical debt, and postpone cleanups forever.